integration_tests/test_basic.py

```
@@ -217,3 +223,176 @@ def test_statesync(cronos):
+            / "contracts/artifacts/contracts/TestRevert.sol/TestRevert.json",
+            KEYS["community"],
+        ),
+        "greeter_1": Greeter(
+            Path(__file__).parent
+            / "contracts/artifacts/contracts/Greeter.sol/Greeter.json",
+            KEYS["signer1"],
+        ),
+        "greeter_2": Greeter(
+            Path(__file__).parent
+            / "contracts/artifacts/contracts/Greeter.sol/Greeter.json",
+            KEYS["signer2"],
+        ),
+    }
+
+    with concurrent.futures.ThreadPoolExecutor(4) as executor:
+        future_to_contract = {
+            executor.submit(contract.deploy, w3): name
+            for name, contract in contracts.items()
+        }
+
+        assert_receipt_transaction_and_block(w3, future_to_contract)
+
+    # Do Multiple contract calls
+    with concurrent.futures.ThreadPoolExecutor(4) as executor:
+        futures = []
+        futures.append(
+            executor.submit(contracts["test_revert_1"].transfer, 5 * 10 ** 18 - 1)
+        )
+        futures.append(
+            executor.submit(contracts["test_revert_2"].transfer, 5 * 10 ** 18)
+        )
+        futures.append(executor.submit(contracts["greeter_1"].transfer, "hello"))
+        futures.append(executor.submit(contracts["greeter_2"].transfer, "world"))
+
+        assert_receipt_transaction_and_block(w3, futures)
+
+        # revert transaction
+        assert futures[0].result()["status"] == 0
+        # normal transaction
+        assert futures[1].result()["status"] == 1
+        # normal transaction
+        assert futures[2].result()["status"] == 1
+        # normal transaction
+        assert futures[3].result()["status"] == 1
+
+
+def assert_receipt_transaction_and_block(w3, futures):
+    receipts = []
+    for future in concurrent.futures.as_completed(futures):
+        # name = future_to_contract[future]
+        try:
+            data = future.result()
+        except Exception as exc:
+            print("%s" % (exc))
```

why ignore the exception here?